subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -252,7 +253,8 @@ public PathVisitor(MerkleDirectorySnapshotBuilder builder, @Nullable Spec<FileTr
 
         @Override
         public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
-            String name = stringInterner.intern(dir.getFileName().toString());
+            String fileName = Optional.ofNullable(dir.getFileName()).map(Object::toString).orElse("");
```

Comment:
Shouldn't the file name of the root directory on Windows be the drive letter? So if you mounted the drive on `Y:` then the `fileName` should be `Y:`. Since `dir` is already the root, `dir.toString()` would yield the drive letter, right?

For Unix, I guess `""` is good. What does `getFileName()` yield on `/`?

In general, I think that joining "file names" with the file separator when starting at the root of the file system you should actually get an absolute path.

Reply:
> What does `getFileName()` yield on `/`?

Same as for windows, if there are no path fragments, both return `null`

So you're saying instead of `""`, `fileName` should be `/` on Linux/Mac and e.g. `x:` on Windows?

Reply:
I think so, yes.

Reply:
Good idea, done, thanks.